### PR TITLE
Add nd support for tiff export job

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
This PR enables tiff export for nd datasets.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz -> not deployed as corresponding worker version is needed.

### Steps to test:
- Needs to be tested together with the worker of PR: https://github.com/scalableminds/voxelytics/pull/3667
- Setup wk and worker locally
- Test 3d dataset tiff export:
  - Choose your favourite 3D dataset, create an annotation and create a bbox
  - Open the download modal  e.g. via the bbox dropdown menu and try to export the bbox as ome tiff and tiff stack zip
- Test new nd dataset tiff export:
  - Choose your favourite ND dataset, create an annotation and create a bbox
  - Open the download modal  e.g. via the bbox dropdown menu and try to export the bbox as ome tiff and tiff stack zip
- All downloads should work
 
### TODOs:
- [ ] Nothing, hopefully :)

### Issues:
- fixes [#7279](https://github.com/scalableminds/webknossos/issues/7279)

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Considered [common edge cases](../blob/master/.github/common_edge_cases.md) - 2d datasets maybe?
- [x] Needs datastore update after deployment

### CODE REVIEW

**Review by philippotto — COMMENTED** [review]
the frontend lgtm :+1:

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/AxisOrder.scala [hunk 1]
```diff
@@ -97,6 +97,12 @@ case class FullAxisOrder(axes: Seq[Axis]) {
   def permuteIndicesArrayToWk(indices: Array[Int]): Array[Int] =
     arrayToWkPermutation.map(indices(_))
 
+  def toWkLibsDictObject: JsValue =
```
Comment (fm3) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/AxisOrder.scala:100]:
```suggestion
  def toWkLibsDictJson: JsValue =
```

File: util/src/main/scala/com/scalableminds/util/geometry/BoundingBox.scala [hunk 2]
```diff
@@ -70,6 +71,9 @@ case class BoundingBox(topLeft: Vec3Int, width: Int, height: Int, depth: Int) {
     Vec3Int(width, height, depth)
 
   def toLiteral: String = f"${topLeft.x},${topLeft.y},${topLeft.z},$width,$height,$depth"
+
+  def toWkLibsDict: JsObject =
```
Comment (fm3) [inline util/src/main/scala/com/scalableminds/util/geometry/BoundingBox.scala:75]:
```suggestion
  def toWkLibsJson: JsObject =
```

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/models/datasource/AdditionalAxis.scala [hunk 3]
```diff
@@ -10,6 +10,17 @@ case class AdditionalAxis(name: String, bounds: Array[Int], index: Int) {
   lazy val lowerBound: Int = bounds(0)
   lazy val upperBound: Int = bounds(1)
   lazy val highestValue: Int = upperBound - 1
+
+  def enclosingAdditionalCoordinates(additionalCoordinates: Seq[AdditionalCoordinate]): AdditionalAxis = {
```
Comment (fm3) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/models/datasource/AdditionalAxis.scala:14]: Could you add a comment what this is for? I didn’t quite understand it on first read. Also, if it returns an Axis, why is it called enclosingAdditionalCoordinates? :thinking:

**Review by fm3 — COMMENTED** [review]
I just tried testing this, but the »Export« button didn’t have any effect for me on this branch. Also no console logging. Works on master branch. Can you reproduce that? I’m on the l4_sample dataset

**Comment by MichaelBuessemeyer** [comment]
> Can you reproduce that?
Nope sorry :see_no_evil:. I just tried again and it works for me with the l4_sample dataset. Maybe your bounding box was special regarding dimensions or `yarn enable-jobs` was missing or the worker not running?

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/AxisOrder.scala (hunk 1 above)
Comment (MichaelBuessemeyer) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/AxisOrder.scala:100]: I also went for `toWkLibsJson` here as below to have the naming more consistent. Feel free to argue

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/models/datasource/AdditionalAxis.scala (hunk 3 above)
Comment (MichaelBuessemeyer) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/models/datasource/AdditionalAxis.scala:14]: I renamed the method and added comments. Is it clearer now?

**Comment by fm3** [comment]
Huh, I re-tried and now it worked :thinking: Not sure what was different. Let’s assume it was a misconfiguration on my part.

One more thing I noticed: the bbox in the job list view at `http://localhost:9000/jobs` is no longer written out correctly.

By the way, I think 2d datasets should not be a problem (as they are handled as 3d with height=1 in wk)

**Comment by MichaelBuessemeyer** [comment]
> One more thing I noticed: the bbox in the job list view at http://localhost:9000/jobs is no longer written out correctly.

Thanks for the hint. Should be fixed now.

I also checked for backwards compatiblity that bboxes of old export jobs can be rendered.
image
The two top most jobs were made on the current master and the lower ones on this branch. Both render with the code of this branch

**Review by fm3 — APPROVED** [review]

**Pull request merged**